appinventor/components/src/com/google/appinventor/components/runtime/SoundRecorder.java

```diff
@@ -158,26 +159,22 @@ public void SavedRecording(String pathName) {
    */
   @SimpleFunction
   public void Start() {
-      // Need to check if we have RECORD_AUDIO permission
+    // Need to check if we have RECORD_AUDIO and WRITE_EXTERNAL permissions
     if (!havePermission) {
       final SoundRecorder me = this;
       form.runOnUiThread(new Runnable() {
-          @Override
-          public void run() {
-            form.askPermission(Manifest.permission.RECORD_AUDIO,
-              new PermissionResultHandler() {
-                @Override
-                public void HandlePermissionResponse(String permission, boolean granted) {
-                  if (granted) {
-                    me.havePermission = true;
-                    me.Start();
-                  } else {
-                    form.dispatchPermissionDeniedEvent(me, "Start", Manifest.permission.RECORD_AUDIO);
-                  }
-                }
-              });
-          }
-        });
+        @Override
+        public void run() {
+          form.askPermission(new BulkPermissionRequest(me, "AudioRecord",
```

Comment:
The second argument here is the name of calling function, so it should be "Start", not "AudioRecord".